src/js/src/parser.cpp

```diff
@@ -0,0 +1,199 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "include/parser.hpp"
+
+#include <future>
+
+#include "include/addon.hpp"
+#include "include/helper.hpp"
+
+JSParser::JSParser(Napi::Env env, Napi::Object jsParser) {
+    js_parser_ref = Napi::Persistent(jsParser);
+    js_parser_ref.SuppressDestruct();
```

Comment:
Is `SuppressDestruct` needed here? 
 `ObjectReference`  without suppressDestruct is used as class property in openvino and it's not destructed too early.